src/components/admin/SemesterDetailCard.tsx

```diff
@@ -0,0 +1,69 @@
+"use client";
+
+import { createContext, memo } from "react";
+import Link from "next/link";
+import { Ellipsis } from "lucide-react";
+
+import { Card } from "@/components/Card";
+import { Popover, PopoverContainer, PopoverOpenButton } from "../popover";
+import DeleteButton from "./DeleteButton";
+import { EditButton } from "./EditButton";
+
+type DetailCardProps = {
+  id: number;
+  name: string;
+  startDate: string;
+  endDate: string;
+  breakStart: string;
+  breakEnd: string;
+};
+export const SemesterContext = createContext<DetailCardProps>(
+  {} as DetailCardProps,
+);
+
+const UnmemoizedSemesterDetailCard = ({
+  id,
+  name,
+  startDate,
+  endDate,
+  breakStart,
+  breakEnd,
+}: DetailCardProps) => {
+  return (
+    <SemesterContext.Provider
+      value={{ id, name, startDate, endDate, breakStart, breakEnd }}
+    >
+      <Card className=" relative bg-secondary/20 ring-1 tracking-tight font-medium ring-secondary text-tertiary text-sm select-none">
+        <Link
+          href={`semesters/${id}`}
+          className="absolute top-0 left-0 w-full h-full z-10"
+        />
+        <div className="flex items-center justify-between">
+          <h3 className="text-lg text-foreground whitespace-nowrap">{name}</h3>
+          <SemesterPopover />
+        </div>
+        <p className="mt-2">Start date: {startDate}</p>
+        <p>End date: {endDate}</p>
+        <p className="mt-4">
+          Break period: {breakStart} - {breakEnd}
+        </p>
+      </Card>
+    </SemesterContext.Provider>
+  );
+};
+
+const SemesterPopover = () => {
+  return (
+    <Popover>
+      <PopoverOpenButton variant="outline" className="w-8 h-6 z-10">
+        <Ellipsis className="stroke-tertiary absolute w-4" />
+      </PopoverOpenButton>
+      <PopoverContainer>
+        <EditButton />
+        <DeleteButton />
+      </PopoverContainer>
+    </Popover>
+  );
+};
```

Comment:
re: making this reusuable instead of creating new Popover / Edit / Delete buttons everytime we need it.

I think you can come back to this after you fully finish the edit and delete implementation

but an idea is passing the form as a prop